ui/lib/sync/addon/routes/secrets.ts

```diff
@@ -8,43 +8,27 @@ import { service } from '@ember/service';
 
 import type RouterService from '@ember/routing/router-service';
 import type StoreService from 'vault/services/store';
-import { DEBUG } from '@glimmer/env';
-
-interface ActivationFlagsResponse {
-  data: {
-    activated: Array<string>;
-    unactivated: Array<string>;
-  };
-}
+import type VersionService from 'vault/services/version';
+import type PersonaService from 'vault/services/persona';
 
 export default class SyncSecretsRoute extends Route {
   @service declare readonly router: RouterService;
   @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+  @service declare readonly persona: PersonaService;
 
-  async fetchActivatedFeatures() {
-    // The read request to the activation-flags endpoint is unauthenticated and root namespace
-    // but the POST is not which is why it's not in the NAMESPACE_ROOT_URLS list
-    return await this.store
-      .adapterFor('application')
-      .ajax('/v1/sys/activation-flags', 'GET', { unauthenticated: true, namespace: null })
-      .then((resp: ActivationFlagsResponse) => {
-        return resp.data?.activated;
-      })
-      .catch((error: unknown) => {
-        if (DEBUG) console.error(error); // eslint-disable-line no-console
-        return [];
-      });
+  beforeModel() {
+    return this.version.fetchActivatedFeatures();
```

Comment:
does this line imply that if we don't call `this.version.fetchActivatedFeatures()` before calling `this.persona.secretsSyncPersona` that `this.persona.secretsSyncPersona` will fail? if so we may want to add some guardrails to prevent us from forgetting to call one before the other.

Reply:
Good reading—yep that's what it does. I noticed this was the pattern for the version service. It has to call the license features endpoint before it returns the getters elsewhere in the app, so it does this in a beforeModel hook on the auth route, plus some other areas if it needs to refresh.

And good call on the guardrails. Let me add that as a todo and revisit.

Reply:
@noelledaley I've played around with this, and I'm not really sure how to go about this. I can't add the request inside the getter otherwise it's fired off every time we access the getter. And I can't really run a check inside the getter either for activatedFeatures length because it could be empty. Thoughts?

We run into this same issue with the version service's `fetchFeatures`. There are no guardrails, instead we just make sure it's called in the beforeModel hooks of the routes that rely on the service.

Reply:
makes sense! definitely not worth making the request every time the getter is called. i started thinking about whether we could add some kind of `assert` inside the `secretSyncPersona` call to make sure `this.version.activatedFeatures` isn't `undefined`, but i'm not sure if that would work either.

i can see `fetchActivatedFeatures` needing to be fetched from a high-level route anyway so we may not even run into the issue. i was just thinking ahead, trying to make sure we don't unintentionally think that a user doesn't have the feature activated when really it's that we haven't checked yet. 😅